scenarios/httpclient.benchmarks.yml

```diff
@@ -0,0 +1,228 @@
+﻿# Local run:
+# crank --config "C:\Users\knatalia\dev\git\Benchmarks\scenarios\httpclient.benchmarks.yml" --scenario httpclient-kestrel-get --profile local
+
+imports:
+  - https://github.com/aspnet/Benchmarks/blob/main/scenarios/aspnet.profiles.yml?raw=true
+  - https://raw.githubusercontent.com/dotnet/crank/main/src/Microsoft.Crank.Jobs.Wrk/wrk.yml
+
+variables:
+    warmup: 15 #seconds -- default value *copied* to WRK, see notes below
+    duration: 30 #seconds -- default value *copied* to WRK
+    serverPort: 1234 # -- default value *copied* to WRK
+    httpVersion: "1.1"
+    useHttps: false
+    numberOfHttpClients: 1
+    concurrencyPerHttpClient: 1 # unsupported by WRK -- will be ignored
+    collectRequestTimings: false # unsupported by WRK -- will be ignored
+
+    # HttpClient-specific parameters:
+    http11MaxConnectionsPerServer: 0 #unlimited
+    http20EnableMultipleConnections: true
+    useWinHttpHandler: false
+    useHttpMessageInvoker: false
+
+scenarios:
+  httpclient-kestrel-get:
+    server:
+      job: kestrel
+    client:
+      job: httpClient
+      variables:
+        scenario: get
+  wrk-kestrel-get:
+    server:
+      job: kestrel
+    client:
+      job: wrk
+      variables:
+        connections: '{{numberOfHttpClients}}'
```

Comment:
Isn't it equivalent to `concurrencyPerHttpClient` instead? 
A single instance of `HttpClient` is able to pool multiple connections, which as assume is equal to `concurrencyPerHttpClient`? This would be what `connections` means for wrk.

Reply:
I don't think so. The terminology in my mind is as follows: *HTTP client* is an entity that allows you to send HTTP requests, and several *HTTP Client*s will be fully independent one from another. In that sense, `System.Net.Http.HttpClient` is an *HTTP client*, but the future LLHTTPConnection would also be an *HTTP client*, and WRK's connection is also an *HTTP client*. 

That way, `numberOfHttpClients` would be a number of guaranteedly independent entities. And `concurrencyPerHttpClient` would be a number of threads accessing a single entity, and whether it supports concurrent access and how well it supports it is up to the implementation and configuration, and could/should be benchmarked. Note that `System.Net.Http.HttpClient` can be configured in a way that it will have only 1 connection, so all the threads will be competing to access it. And even if not limited, there is no way to make the connections inside `HttpClient` fully independent.

Unfortunately, WRK doesn't seem to support accessing one connection from several threads (it was more about handling several connections in one thread). That's why I set number of threads in WRK to always be equal to number of connections.

Reply:
> Note that System.Net.Http.HttpClient can be configured in a way that it will have only 1 connection

In the PR, `numberOfHttpClients` is used for the `connections` property in wrk. When set it to `1` it will create exactly 1 connection to the server, with one wrk thread, so I assume a max concurrency level of 1. However the default value for `http11MaxConnectionsPerServer` is "unlimited". Running it with HttpClient should create `concurrencyPerHttpClient` physical connections at max. Why use the same variables in wrk when they don't represent the same thing, while leaving `concurrencyPerHttpClient` out of wrk when this impact how many connections can be created. Why not set wrk 's `connections` property to `numberOfHttpClients | times:  concurrencyPerHttpClient` (the syntax in liquid for multiplications)? Even if the threading model is different and could be a fixed value based on the numbers of cores of the machine (it's available in the profiles). TE for instances fixes it to 32 thread.

I would assume that the wrk scenario is defined to have a baseline for httpclient, but if one scenario create 200 connections with a set of variables, and the other a single one, I don't see how it's fair.

Reply:
I agree that currently it will be wrong to call WRK when `concurrencyPerHttpClient` is not 1. Having the value to be `numberOfHttpClients` x `concurrencyPerHttpClient` will be great actually. But I wasn't able to make it work the way you suggest (I get parser errors). I will reach out to you offline.

Reply:
(other way could be to have some kind of validation that concurrencyPerHttpClient=1, I wonder if it is possible with the syntax)

Reply:
// covered that offline, I'll do the change